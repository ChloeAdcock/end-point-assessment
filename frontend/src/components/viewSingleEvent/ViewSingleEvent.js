import React, { useState, useEffect } from 'react';
import Typography from '@material-ui/core/Typography';
import MapContainer from '../mapContainer/MapContainer';
import { addressFromLatLong } from '../../redux/actions/geocoding/geocoding';
import formatDateTime from '../../helpers/formatDateTime';

function ViewSingleEvent(props) {

    const [address, setAddress] = useState("");
    const [selectedEvent] = useState(props.history.location.state);
    const [eventArray, setEventArray] = useState([]);

    useEffect(() => {
        async function getFormattedAddress() {
            const formattedAddress = await addressFromLatLong(selectedEvent.latitude, selectedEvent.longitude);
            setAddress(formattedAddress);
        }
        getFormattedAddress();
        setEventArray(oldArray => [...oldArray, selectedEvent]);
    }, [selectedEvent]);

    const mapStyles = {
        height: "50vh",
        width: "50%"
    };

    if (!eventArray || !address) {
        return <Typography>Loading...</Typography>
    } else {
        return (
            <div>
                <Typography variant='h2'>{selectedEvent.name}</Typography>
                <Typography variant='h6'>{formatDateTime(selectedEvent.date_time)}</Typography>
                <Typography variant='body1'>For more information contact {selectedEvent.contact_info}</Typography>
                <Typography variant='body1'>{selectedEvent.description}</Typography>
                <Typography variant='body1'>{address}</Typography>
                <MapContainer events={eventArray} mapStyles={mapStyles} centre={{
                    lat: Number(selectedEvent.latitude), lng: Number(selectedEvent.longitude)
                }} />
            </div>
        )
    }
}

export default ViewSingleEvent;