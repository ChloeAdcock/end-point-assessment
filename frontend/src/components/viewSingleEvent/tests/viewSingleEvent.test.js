import renderer from "react-test-renderer";
import ViewSingleEvent from "../ViewSingleEvent";

describe("The view single event component", () => {
  it("should render loading if there is nothing the event array", () => {
    const historyMock = {
      location: {
        state: {},
      },
    };
    const tree = renderer
      .create(<ViewSingleEvent history={historyMock} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
