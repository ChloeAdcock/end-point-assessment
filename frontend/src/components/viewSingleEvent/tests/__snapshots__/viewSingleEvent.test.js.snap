// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The view single event component should render loading if there is nothing the event array 1`] = `
<div
  className="MuiContainer-root makeStyles-message-3 MuiContainer-maxWidthLg"
>
  <h5
    className="MuiTypography-root MuiTypography-h5"
  >
    Loading...
  </h5>
  ;
</div>
`;
